executor/write.go

```diff
@@ -279,6 +280,9 @@ func (e *DeleteExec) deleteSingleTable(goCtx goctx.Context) error {
 	batchSize := e.ctx.GetSessionVars().DMLBatchSize
 	for {
 		if batchDelete && rowCount >= batchSize {
+			if err := e.ctx.Txn().FlushStatement(true); err != nil {
```

Comment:
How about call e.ctx.Txn().FlushStatement(true) in e.ctx.NewTxn()? Avoid missing Flush.

Reply:
ditto